### PR TITLE
Add APICurio guide for modifying and visualising kuadrant extensions

### PR DESCRIPTION
**Pull request by david-martin**
Closes https://github.com/Kuadrant/docs.kuadrant.io/issues/63

### CODE REVIEW

File: doc/openapi-apicurio.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+## Usin APICurio to edit an OpenAPI spec and visualise Kuadrant extensions
```
Comment (jasonmadigan) [inline doc/openapi-apicurio.md:1]:
```suggestion
## Using APICurio to edit an OpenAPI spec and visualise Kuadrant extensions
```

File: doc/openapi-apicurio.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            labels:
+                deployment: petstore
+                owner: jbloggs
+            hostnames:
+                - 'petstore.example.com'
+            parentRefs:
+                -   name: prod-web
+                    namespace: kuadrant-multi-cluster-gateways
+                    kind: Gateway
+```
```
Comment (jasonmadigan) [inline doc/openapi-apicurio.md:29]:
```suggestion
```yaml
info:
    x-kuadrant:
        route:
            name: petstore
            namespace: petstore
            hostnames:
                - 'petstore.example.com'
            parentRefs:
                -   name: prod-web
                    namespace: kuadrant-multi-cluster-gateways
                    kind: Gateway
```
```

Comment (jasonmadigan) [reply doc/openapi-apicurio.md:29]: can probably remove the labelling stuff

File: doc/openapi-apicurio.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+## Usin APICurio to edit an OpenAPI spec and visualise Kuadrant extensions
+
+This guide assumes you have APICurio already running.
+See https://www.apicur.io/ for info on how to install APICurio Studio.
+
+### Editing the spec
```
Comment (jasonmadigan) [inline doc/openapi-apicurio.md:6]: Before diving into editing, might be worth a pre-amble on OAS extensions, linking out to https://swagger.io/docs/specification/openapi-extensions/

File: doc/openapi-apicurio.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            in: header
+```
+
+When added, the UI will display this in the security requirements section:
+
+![APICurio security requirements](./images/apicurio-security-scheme-apikey.png)
+
+See [this guide](./generate-kuadrant-auth-policy.md) for more info on generating an AuthPolicy.
+
+To generate a RateLimitPolicy for the API, add the following `x-kuadrant` block to a path in your spec,
```
Comment (jasonmadigan) [inline doc/openapi-apicurio.md:54]:
```suggestion
To generate a `RateLimitPolicy` for the API, add the following `x-kuadrant` block to a path in your spec,
```

Comment (jasonmadigan) [reply doc/openapi-apicurio.md:54]: perhaps also a link to the RLP API dcs.

**Review by jasonmadigan — CHANGES_REQUESTED** [review]

**Comment by jasonmadigan** [comment]
taking shape, couple of comments added

**Comment by jasonmadigan** [comment]
@david-martin this LGTM. It may be worth an eye from someone from Apicurio as well

**Comment by david-martin** [comment]
> may be worth an eye from someone from Apicurio

@jasonmadigan Good suggestion.
Do you mind if you add lgtm etc.. and I can get a deploy preview link on the docs repo to share with them for review before it actually gets published?

**Comment by david-martin** [comment]
Merging manually. 
Auto merge not marking the GH review as good, but lgtm added

**Pull request merged**